internal/orchestrator/updates.go

```diff
@@ -485,24 +629,69 @@ func (o *Orchestrator) startReconciler(ctx context.Context, wg *sync.WaitGroup)
 				keepRunning = false
 
 			case <-ticker.C:
-				evts := o.getEventsToReconcile(ctx)
-				for _, evt := range evts {
-					le := o.logger.WithFields(logrus.Fields{
-						"conditionID":    evt.ConditionUpdate.ConditionID.String(),
-						"conditionState": string(evt.ConditionUpdate.State),
-						"kind":           string(evt.Kind),
-					})
-
-					if err := o.eventUpdate(ctx, evt); err != nil {
-						le.WithError(err).Warn("reconciler event update")
-						continue
-					}
-
-					if err := o.notifier.Send(evt); err != nil {
-						le.WithError(err).Warn("reconciler event notification")
-					}
-				}
+				o.reconcileStatusKVEntries(ctx)
+				o.reconcileActiveConditionRecords(ctx)
 			}
 		}
 	}()
 }
+
+func (o *Orchestrator) reconcileStatusKVEntries(ctx context.Context) {
+	evts := o.getEventsToReconcile(ctx)
+	for _, evt := range evts {
+		le := o.logger.WithFields(logrus.Fields{
+			"conditionID":    evt.ConditionUpdate.ConditionID.String(),
+			"conditionState": string(evt.ConditionUpdate.State),
+			"kind":           string(evt.Kind),
+		})
+
+		if err := o.eventUpdate(ctx, evt); err != nil {
+			le.WithError(err).Warn("reconciler event update")
+			continue
+		}
+
+		le.Info("condition reconciled")
+
+		if err := o.notifier.Send(evt); err != nil {
+			le.WithError(err).Warn("reconciler event notification")
+		}
+	}
+}
+
+// reconcile active-condition KV entries for missing status KV entries and missing active-condition entries
+func (o *Orchestrator) reconcileActiveConditionRecords(ctx context.Context) {
```

Comment:
I feel like separating this process into `creates` and `updates` has the potential to miss things, like notifications. Why not conditionally create the `active_condition` record if the error from `UpdateCondition` is `ErrConditionNotFound`?

Reply:
The notification was excluded on purpose here, since the `Notifier.Send()` expects a `ConditionUpdateEvent` and `creates` is a slice of `Condition` types, while `updates` is a slice of `ConditionUpdateEvents`.
 This is the main reason that they are individually handled as well - because they are different types and `UpdateCondition()` expects the `ConditionUpdateEvent` type.